packages/coinstac-pipeline/src/pipeline-manager.js

```diff
@@ -541,6 +541,11 @@ module.exports = {
 
         return this.activePipelines[runId].stateEmitter;
       },
+      stopPipeline(pipelineId, runId) {
+        this.activePipelines[runId].pipeline.pipelineSteps.forEach( step => {
```

Comment:
Couple things here, the pipeline-manager here is directly mussing w/ pipeline objects, I try to do this only when absolutely necessary. Also, there is only ever one currently running pipeline `step` and so you would only need to ever stop that one. 
What needs to happen is that the call to stop the currently running step needs to happen from manager to a pipeline function, and that function needs to call the currently active step's controller, thus stopping the pipeline.

Reply:
Also also, docbloc your functions :-)